pkg/fqdn/regexpmap/regexp_map.go

```diff
@@ -111,24 +222,28 @@ func (m *RegexpMap) LookupContainsValue(lookupKey, expectedValue string) (found
 // lookupValues remain for reStr the internall regexp is deleted (later Adds
 // will recompile it).
 func (m *RegexpMap) Remove(reStr, lookupValue string) (deleted bool) {
-	if _, exists := m.rules[reStr]; !exists {
-		return false
+	lookupRelation, exists := m.rulesRelation[reStr]
+	if !exists {
+		delete(m.rules, reStr)
+		return true
```

Comment:
I'm not sure that this ended up being critical to te new code working, but this should probably return false (to match what it did before). The `delete(m.rules, reStr)` will also be confusing so we may as well not do it (I mean, we should correctly delete it below and we do)